### ISSUE TITLE
Binary dump of traffic with 50 HP

### ISSUE THREAD

**Issue by phdussud** [dulitz/finitude#4]
@dulitz Hello, 
My HP failed and it looks like there is no replacement available for it. I am left with a choice of non ICS HPs. I would like to build a board to emulate the ICS HP traffic and control the new HP. I have built some HW to sniff the traffic and use your code to sniff the traffic but since my HP is offline, I don't get to sniff the traffic to it. If I could get my hands on a binary dump that contains the basic commands of the HP (identification, on, off, stage1, stage2) I feel I can build the board I want. 
Thank you very much. 
P.S I like your code, it is simple and comprehensive. 
Patrick

**Comment by dulitz** [comment]
Sorry for the delay Patrick, and thank you for your compliment. Are you still interested in this?

It's a bit more complicated than one would hope, because the thermostat issues various probes to find the heat pump and many of those we don't know what we do.

What I think we can do is figure out how to emulate the NIM, which is the Carrier box that interfaces ABCD to a unit that is controlled by a traditional analog thermostat. (Or you could buy a NIM, but emulation is probably better.) If you're still interested in this I can give you a dump.

Again apologies for not being around for this.

**Comment by phdussud** [comment]
Hello!
Thank you for your thoughtful response. I was able to purchase one of the last compressor available and a nice independent contractor was willing to install it on my heat pump. The heat pump is now functional and I have dumps of the traffic to and from it. As you said there are tons of opaque data.
I do have a NIM that serves as the interface to my fresh air system and I wondered if it could serve the both fresh air system and a dumb heat pump. The installation manual does not say anything about dual use, but the connections are separate so it's only limited by the firmware of the NIM.
Exploring the NIM or a clone of it was my backup plan.
Long story short, I don't need anything anymore. I have a ESP32 logger that buffers (SD card) the binary logs until I pull them out to my desktop. I have modified version of finitude (which I found well thought out and easy to work with) which reads the logs and dumps the decoded info into a SLQ database.
Thanks again for contacting me.
Patrick

________________________________

**Comment by dulitz** [comment]
Glad you got your problem solved! My NIM is connected to an HRV (4-wire cable entering the top left of the NIM) and a non-communicating air conditioner (cable entering the bottom left of the NIM) so it can do both.

Would you send a PR for your database dump code? Maybe someone else would find it useful.

**Comment by phdussud** [comment]
That's a good idea. I will prepare a PR as soon as I have a free block of time.

Patrick
________________________________
